shop4All-backend/src/main/java/com/example/shop4All_backend/service/JwtService.java

```diff
@@ -79,48 +64,37 @@ else if (seller.isPresent()) {
     // Load user details by username (email in this case)
     @Override
     public UserDetails loadUserByUsername(String userEmail) throws UsernameNotFoundException {
-        Optional<Admin> admin = adminRepo.findById(userEmail);
-        Optional<Seller> seller = sellerRepo.findById(userEmail);
-        Optional<Buyer> buyer = buyerRepo.findById(userEmail);
-
-        User user;
-        if (admin.isPresent()) {
-            user = admin.get();
-        }
-        else if (seller.isPresent()) {
-            user = seller.get();
-        }
-        else{
-            user = buyer.get();
-        }
-
-        if (user != null){
+        User user = userRepo.findByUserEmail(userEmail).orElse(null);
+        if (user != null) {
             return new org.springframework.security.core.userdetails.User(
-              user.getUserEmail(),user.getUserPassword(),getAuthorities(user)
+                    user.getUserEmail(), user.getUserPassword(), getAuthorities(user)
             );
-        }
-        else{
-            throw new UsernameNotFoundException("User not found");
+        } else {
+            throw new LogInException("User not found");
         }
     }
 
     // Get authorities (roles) for the user
     private Set getAuthorities(User user) {
-        Set authorities = new HashSet();
-
-        user.getRole().forEach(role -> authorities.add(new SimpleGrantedAuthority("ROLE_"+role.getRoleName())));
-
+        Set<SimpleGrantedAuthority> authorities = new HashSet<>();
+        Role role = user.getRole();
+        authorities.add(new SimpleGrantedAuthority("ROLE_" + role.name()));
         return authorities;
     }
 
+
     // Authenticate user credentials
-    private void authenticate(String userEmail, String userPassword) throws Exception {
+    private void authenticate(String userEmail, String userPassword) throws AuthenticationException {
         try {
             authenticationManager.authenticate(new UsernamePasswordAuthenticationToken(userEmail, userPassword));
         } catch (DisabledException e) {
-            throw new Exception("User is disable");
-        } catch (BadCredentialsException e){
-            throw new Exception("Bad credential for user");
+            throw new LogInException("User is disabled");
+        } catch (BadCredentialsException e) {
+            throw new LogInException("Bad credentials for user");
+        } catch (AuthenticationException e) {
+            throw new LogInException("Authentication failed");
```

Comment:
Did you tested if all the exceptions are called if something went wrong? I am asking because you have to make sure that the order of the Exceptions classes is okay so that all the catch block will be executed if you get an exception, not just one block.

For example, you should put the Exception catch block last because every other exceptions are derived from Exception.